tests/integration/_app.typegen.ts

```diff
@@ -97,6 +97,29 @@ export interface NexusGenFieldTypes {
   }
 }
 
+export interface NexusGenFieldTypeNames {
+  Mutation: {
+    // field return type
+    createUser: 'User'
+  }
+  Post: {
+    // field return type
+    body: 'String'
+    title: 'String'
+  }
+  Query: {
+    // field return type
+    foo: 'String'
+    searchPosts: 'Post'
+    user: 'User'
+  }
+  User: {
+    // field return type
+    firstName: 'String'
+    lastName: 'String'
+  }
+}
```

Comment:
New interface added to typegen which is used to make the `ResolverReturnTypeName` typegen helper, used to type the connection field extension